Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,61 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _security_identities:
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint32``,
+which means the maximum limit for a security identity is ``2^32 - 1``. The
+minimum security identity is ``1``, due to the reserved identity range being
+from ``1`` to ``255``. See the `internal code documentation
+<https://pkg.go.dev/github.com/cilium/cilium/pkg/identity#NumericIdentity>`_.
+
+.. note::
+
+   Identitiy 0 is not a valid value. If it shows up in Hubble output, this
+   means the identity was not found. In the eBPF datapath, it has a special
+   role where it denotes "any identity", i.e. as a wildcard allow in policy
+   map.
+
+There are two caveats to the above however:
+
+1) ClusterMesh
+2) Identities generated from CIDR-based policies
+
+For (1), 8 bits are used as the ``cluster-id`` which identifies the cluster in
+the ClusterMesh, into the 3rd octet as shown by ``0x00FF0000``. This does not
+apply to CIDR identities however, see (2).
+
+For (2), CIDR identities are local to each node. This means that they're not
+subject to the global limit of security identities (``2^16 - 1``). CIDR
```

Comment:
This “global limit” is no longer defined above, security identities are now introduced as `uint32`s.